File: projects/stock/src/main/java/com/vjcspy/spring/packages/stockinfo/dto/CorEntityDto.java
/* (C) 2024 */
package com.vjcspy.spring.packages.stockinfo.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.math.BigInteger;
import java.time.LocalDate;
import lombok.Data;

@Data
public class CorEntityDto {
    private Integer id;
    private Integer refId;
    private Integer catId;
    private String code;
    private String exchange;
    private String industryName1;
    private String industryName2;
    private String industryName3;
    private BigInteger totalShares;
    private String name;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy")
    private LocalDate firstTradeDate;
}

File: projects/stock/src/main/kotlin/com/vjcspy/spring/packages/stockinfo/dto/CorEntityDto.kt
// (mr.vjcspy@gmail.com) 2024
package com.vjcspy.spring.packages.stockinfo.dto

import com.fasterxml.jackson.annotation.JsonFormat
import java.math.BigInteger
import java.time.LocalDate

data class CorEntityDto(
    val id: Int? = null,
    val refId: Int? = null,
    val catId: Int? = null,
    val code: String? = null,
    val exchange: String? = null,
    val industryName1: String? = null,
    val industryName2: String? = null,
    val industryName3: String? = null,
    val totalShares: BigInteger? = null,
    val name: String? = null,
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd-MM-yyyy")
    val firstTradeDate: LocalDate? = null,
)
